### DISCUSSION TITLE
Create Schedule within the context of a Transaction.

### DISCUSSION THREAD

**Discussion by kartomic8** [timgit/pg-boss#388] [Q&A] [answered]
I'm trying to create a schedule in the context of a transaction.  If the transaction gets rolledback, I don't want pgboss to create the schedule.  However, it seems that the db option for schedule() doesn't get considered.

Is this by design, or is this a bug / missing feature?

**Comment by timgit** [comment, marked as answer]
This is a missing feature

**Reply by SylvainMarty** [reply]
Hi @timgit and thanks for this library, it's really awesome!
Sadly, the fact we can't easily wrap PGBoss inserts into an existing transaction is really a miss.
To me, PGBoss strength is that it leaves with the existing database while providing exactly-once job processing without having to implement change data capture system or any heavy duty infrastructure.
It should also provides strong ACID capabilities to actually guarantee exactly-once delivery from the creation to the processing of the job.

I would love to give it a try and contribute but since this feature has been missing for a while, I'm not sure why it hasn't been implemented yet...
Are you interested to see this feature implemented in PGBoss?
Also, there was an attempt to implement this years ago: https://github.com/timgit/pg-boss/pull/200.
Do you remember why this PR has not been completed?

**Reply by timgit** [reply]
It's still missing. :)  It would easy to open a PR to add this if you're up for it.

**Reply by SylvainMarty** [reply]
Okay I'm gonna look into this when I have some time. 👍

**Reply by SylvainMarty** [reply]
@timgit I submitted a PR : https://github.com/timgit/pg-boss/pull/515
